src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12205,35 +12230,39 @@ LambdaExpressionSyntax parseLambdaExpressionWorker()
                 else
                 {
                     var name = this.ParseIdentifierToken();
-                    SyntaxToken arrow, exclamation;
-                    // Case x! =>
+                    SyntaxToken arrow, exclamationExclamation;
+                    // Case x!! =>
                     if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
                     {
-                        exclamation = this.EatToken(SyntaxKind.ExclamationToken);
+                        exclamationExclamation = this.EatToken();
+                        if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
+                        {
+                            // no trailing trivia and no leading trivia
+                            if (exclamationExclamation.GetTrailingTriviaWidth() == 0 && this.CurrentToken.GetLeadingTriviaWidth() == 0)
+                            {
+                                var exclamation2 = this.EatToken();
+                                exclamationExclamation = SyntaxFactory.Token(exclamationExclamation.GetLeadingTrivia(), SyntaxKind.ExclamationExclamationToken, exclamation2.GetTrailingTrivia());
+                            }
+                        }
+                        else
+                        {
+                            exclamationExclamation = this.AddError(exclamationExclamation, ErrorCode.ERR_IncorrectNullCheckSyntax);
```

Comment:
Same here: nothing fancy needed. Just do a regular `EatToken(SyntaxKind.ExclamationToken)`, and an appropriate error will be added. Then we can unconditionally glue those together.